modules/phase_field/src/ics/CoupledValueFunctionIC.C

```diff
@@ -0,0 +1,50 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CoupledValueFunctionIC.h"
+#include "Function.h"
+
+registerMooseObject("PhaseFieldApp", CoupledValueFunctionIC);
+
+InputParameters
+CoupledValueFunctionIC::validParams()
+{
+  InputParameters params = InitialCondition::validParams();
+  params.addClassDescription("Initialize the variable from a lookup function");
+  params.addRequiredParam<FunctionName>("function",
+                                        "Coupled function to evaluate with values from v");
+  params.addCoupledVar("v",
+                       "List of up to four coupled variables that are substituted for x,y,z, and t "
+                       "in the coupled function");
```

Comment:
I'm a little concerned about putting these in a specific order. 

Is there no case where one would need to provide only x, y, and t? Would seem silly to have to provide a dummy for z.

Reply:
Hm, the application I have in mind is creating a special mapping function for a n variable problem. For a three variable problem you'd just always use x,y,z. Making that configurable seems like unnecessary complexity.

Reply:
Alright. That's fine with me, as long as it was kept in mind with the design. 

I'm good to approve once the doc changes are added and docs pass.